ui/src/lib/gql/workspace/useQueries.ts

```diff
@@ -14,35 +20,48 @@ export const useQueries = () => {
   const queryClient = useQueryClient();
 
   const createNewWorkspaceObject = useCallback(
-    (w: WorkspaceFragment): Workspace => ({
-      id: w.id,
-      name: w.name,
-      personal: w.personal,
-    }),
+    (w: WorkspaceFragment | undefined): Workspace | undefined =>
+      w
+        ? {
+            id: w.id,
+            name: w.name,
+            personal: w.personal,
+            members: w.members.map(
+              (m): Member => ({
+                userId: m.userId,
+                role: m.role,
+                user: m.user
+                  ? {
+                      id: m.user?.id,
+                      name: m.user?.name,
+                      email: m.user?.email,
+                    }
+                  : undefined,
+              }),
+            ),
+          }
+        : undefined,
     [],
   );
 
   const createWorkspaceMutation = useMutation({
     mutationFn: async (name: string) => {
       const data = await graphQLContext?.CreateWorkspace({ input: { name } });
-      return (
-        data?.createWorkspace?.workspace &&
-        createNewWorkspaceObject(data?.createWorkspace?.workspace)
-      );
+      return createNewWorkspaceObject(data?.createWorkspace?.workspace);
     },
-    onSuccess: createdWorkspace => {
-      queryClient.setQueryData([WorkspaceQueryKeys.GetWorkspaces], (data: Workspace[]) => [
-        ...data,
-        createdWorkspace,
-      ]);
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: [WorkspaceQueryKeys.GetWorkspaces],
+      });
     },
   });
 
   const useGetWorkspacesQuery = () =>
     useQuery({
       queryKey: [WorkspaceQueryKeys.GetWorkspaces],
       queryFn: () => graphQLContext?.GetWorkspaces(),
-      select: data => data?.me?.workspaces.map(w => createNewWorkspaceObject(w)),
+      select: data =>
+        data?.me?.workspaces.map(w => createNewWorkspaceObject(w)).flatMap(w => (w ? [w] : [])),
```

Comment:
flatMap should get rid of all null/undefined items so this should be enough
```suggestion
        data?.me?.workspaces.flatMap(w => createNewWorkspaceObject(w)),
```

Reply:
I know it should work but Typescript seems to disagree. I tried more things but to no avail. with `as` keyword it does work.

Reply:
Wasn't the introduction of flatMap a couple PRs ago the solution to the typescript issue? 🤔

Reply:
Yes it does but that assumes the `createNewWorkspaceObject` will return only `Workspace` but I updated that it can return `undefined` as well so it simplified the code in other places
flatmap would work only if the return of the function is `workspace`